### PR TITLE
Use query string to limit number of comments.

### PR DESCRIPTION
**Pull request by andres-holguin**
Use the query string in the Request to limit the number of comments retrieved from Datastore.

I wasn't sure if getIntParameter() should return -1 or null. It currently returns null but I've heard that may be bad practice. Any suggestions?

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -52,15 +57,33 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
   }
 
   /**
-   * Helper function that returns an Array of strings with all the comments in Datastore
+   * Helper function to get and attempt to parse an integer parameter in request's query string.
```
Comment (nickKelman) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:60]: I think null and -1 both work fine here as long as it is documented in the comment =)

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -52,15 +57,33 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
+
+  /**
+   * Helper function that returns an ArrayList of strings 
+   * with `numComments` of the comments in Datastore.
+   * All comments will be loaded if numComments < 0
    */
-  private ArrayList<String> getComments() {
+  private ArrayList<String> getComments(Integer numComments) {
     ArrayList<String> comments = new ArrayList<String>();
+    if (numComments ==  null || numComments < 0) numComments = Integer.MAX_VALUE;
```
Comment (nickKelman) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:81]: nice edge case handling

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -66,15 +66,28 @@ function onBodyLoad() {
...
     });
   });
+}
+
+/**
+ * Clear comments from comments list.
+ */
+function clearComments() {
+  const commentsEl = document.getElementById("comments-list");
+  if (commentsEl)
```
Comment (nickKelman) [inline portfolio/src/main/webapp/script.js:91]: You shouldn't imply curly braces. Throw them in here

Comment (andres-holguin) [reply portfolio/src/main/webapp/script.js:91]: Gotcha. Is `  if (commentsEl) commentsEl.innerHTML = "";` Ok?

Comment (nickKelman) [reply portfolio/src/main/webapp/script.js:91]: Yes, see here for more info about google's style guide:
go/tsstyle#control-flow-statements-blocks

Comment (andres-holguin) [reply portfolio/src/main/webapp/script.js:91]: Got hit with UberProxy 🤔 but I found this one https://google.github.io/styleguide/jsguide.html

Comment (nickKelman) [reply portfolio/src/main/webapp/script.js:91]: Ah weird. Here is the relevant bit in the guide you linked:

go/tsstyle#control-flow-statements-blocks

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -52,15 +57,33 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
 
   /**
-   * Helper function that returns an Array of strings with all the comments in Datastore
+   * Helper function to get and attempt to parse an integer parameter in request's query string.
+   */
+  private Integer getIntParameter(HttpServletRequest request, String param) {
+    String numberString = request.getParameter(param);
+
+    try {
+      int number = Integer.parseInt(numberString);
```
Comment (cftorres) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:66]: nit: you can just directly return `Integer.parseInt(numberString)` here

**Pull request merged**